tests/e2e/config_secret_utils.go

```diff
@@ -0,0 +1,584 @@
+	for i := 0; i < len(hosts); i++ {
+		setPermissionForHosts := govcLoginCmd() + "govc permissions.set -principal " +
+			testUser + " -propagate=" + propagateVal + " -role ReadOnly " + hosts[i] + "| tr -d '\n'"
+		result, err := sshExec(sshClientConfig, masterIp, setPermissionForHosts)
+		if err != nil && result.Code != 0 {
+			fssh.LogResult(result)
+			return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+				setPermissionForHosts, masterIp, err)
+		}
+	}
+	return nil
+}
+
+// setVMLevelPermission is used to set vm level permissions for test user
+func setVMLevelPermission(masterIp string, testUserAlias string, testUser string, vms []string,
+	propagateVal string) error {
+	for i := 0; i < len(vms); i++ {
+		setPermissionForK8sVms := govcLoginCmd() + "govc permissions.set -principal " +
+			testUserAlias + " -propagate=" + propagateVal + " -role CNS-VM-" + testUser +
+			" " + vms[i] + " | tr -d '\n'"
+		result, err := sshExec(sshClientConfig, masterIp, setPermissionForK8sVms)
+		if err != nil && result.Code != 0 {
+			fssh.LogResult(result)
+			return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+				setPermissionForK8sVms, masterIp, err)
+		}
+	}
+	return nil
+}
+
+// setClusterLevelPermission is used to set cluster level permissions for test user
+func setClusterLevelPermission(masterIp string, testUserAlias string, testUser string, cluster string,
+	propagateVal string) error {
+	setPermissionForCluster := govcLoginCmd() + "govc permissions.set -principal " +
+		testUserAlias + " -propagate=" + propagateVal + " -role CNS-HOST-CONFIG-STORAGE-" +
+		testUser + " " + cluster + " | tr -d '\n'"
+	result, err := sshExec(sshClientConfig, masterIp, setPermissionForCluster)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			setPermissionForCluster, masterIp, err)
+	}
+	return nil
+}
+
+// setDataStoreLevelPermission is used to set datastore level permissions for test user
+func setDataStoreLevelPermission(masterIp string, testUserAlias string, testUser string, dataCenter string,
+	datastores []string, propagateVal string) error {
+	for i := 0; i < len(datastores); i++ {
+		setPermissionForDataStore := govcLoginCmd() + "govc permissions.set -principal " +
+			testUserAlias + " " +
+			"-propagate=" + propagateVal + " -role CNS-DATASTORE-" + testUser + " '" + datastores[i] + "'"
+		result, err := sshExec(sshClientConfig, masterIp, setPermissionForDataStore)
+		if err != nil && result.Code != 0 {
+			fssh.LogResult(result)
+			return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+				setPermissionForDataStore, masterIp, err)
+		}
+	}
+	return nil
+}
+
+// setSearchlevelPermission method is used to set search level permissions
+func setSearchlevelPermission(masterIp string, testUserAlias string, testUser string, propagateVal string) error {
```

Comment:
keep all low level methods generic and don't hard code values in them, you can use a wrapper with hardcoded valued around it